main/store/actions/index.js

```diff
@@ -134,12 +136,17 @@ module.exports = {
       return updated
     })
   },
-  updateAccount: (u, updatedAccount) => {
-    u('main.accounts', updatedAccount.id, account => {
-      // if (account) return updatedAccount // Account exists
-      // if (add) return updatedAccount // Account is new and should be added
-      return { ...updatedAccount, balances: (account || {}).balances }
+  updateAccount: (u, updatedAccount, newName) => {
+    const { id } = updatedAccount
+    u('main.accounts', id, (account = {}) => {
+      return { ...updatedAccount, balances: account.balances || {} }
     })
+    if (newName && !isDefaultAccountName({ ...updatedAccount, name: newName })) {
```

Comment:
do we need this flag? I would think we want to update the account to save the name in all circumstances

Reply:
Two cases here where this does not update the meta, one is that the new name is an empty string.  
The other is the new name is undefined, which happens when the account has not been renamed.

Reply:
for the first case, can we catch it with `if (account.name) ...` ?
for the second, is there any harm in making this update if the account hasn't been renamed? I would like to avoid tying the knowledge about whether or not to update the meta to the calling code if possible, as it makes the code more brittle. for example if we ever introduced a flow in the future where some code updates the account name and then calls this function the updated name wouldn't be saved.

Reply:
Updated.  We can save the empty string too

Reply:
Wow, our tests really are tightly coupled...